question26/js/login.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const submitButton = document.getElementById('js-submit-button');
+const userInfoInputArea = document.getElementById('name_mail');
+const passwordInputArea = document.getElementById('password');
+const form = document.getElementById('form');
+
+if(localStorage.hasOwnProperty('token')) {
+  window.location.href = './newsComponents/index.html'
+}
+
+const isValidStatus = {
+  name_mail: false,
+  password: false,
+}
+
+const validationInfo = {
+  name_mail: {
+    validation: (value) => value
+  },
+  password: {
+    validation : (value) => /(?=.*[a-zA-Z])(?=.*[0-9])[a-zA-z0-9]{8,}/.test(value),
+    errorMessage: '※8文字以上の大小の英数字を交ぜたものにしてください。'  
+  }
+}
+
+const setValidationEvents = (isValid, targetForm) => {
+  if (isValid) {
+    targetForm.classList.add('valid');
+    targetForm.classList.remove('invalid');
+    targetForm.nextElementSibling.textContent = ''; 
+    isValidStatus[targetForm.id] = true;
+  } else {
+    targetForm.classList.add('invalid');
+    targetForm.classList.remove('valid');
+    targetForm.nextElementSibling.textContent = validationInfo[targetForm.id].errorMessage;
+    isValidStatus[targetForm.id] = false; 
+  }
+}
```

Comment:
IMO
I think I can cut out the function here!
```suggestion
const setValidationEvents = (isValid, targetForm) =>
  isValid ? hogehoge(targetForm) : hugahuga(targetForm);

const hogehoge = (targetForm) => {
  targetForm.classList.add("valid");
  targetForm.classList.remove("invalid");
  targetForm.nextElementSibling.textContent = "";
  isValidStatus[targetForm.id] = true;
};

const hugahuga = (targetForm) => {
  targetForm.classList.add("invalid");
  targetForm.classList.remove("valid");
  targetForm.nextElementSibling.textContent =
    validationInfo[targetForm.id].errorMessage;
  isValidStatus[targetForm.id] = false;
};
```

Reply:
Thank you for your ideas! As you said, we were able to split the function! Fixed it so please confirm.

fixed commit: https://github.com/csakyo/homework/pull/40/commits/945dc0732a517ad66f0669f36ead7d5641cb7b64